pkg/build/builder/source.go

```diff
@@ -190,3 +202,88 @@ func extractGitSource(gitClient GitClient, gitSource *api.GitBuildSource, revisi
 	}
 	return true, nil
 }
+
+func copyImageSource(dockerClient DockerClient, containerID, sourceDir, destDir string, tarHelper tar.Tar) error {
+	// Setup destination directory
+	fi, err := os.Stat(destDir)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		glog.V(4).Infof("Creating image destination directory: %s", destDir)
+		err := os.MkdirAll(destDir, 0644)
+		if err != nil {
+			return err
+		}
+	} else {
+		if !fi.IsDir() {
+			return fmt.Errorf("destination %s must be a directory", destDir)
+		}
+	}
+
+	tempFile, err := ioutil.TempFile("", "imgsrc")
+	if err != nil {
+		return err
+	}
+	glog.V(4).Infof("Downloading source from path %s in container %s to temporary archive %s", sourceDir, containerID, tempFile.Name())
+	err = dockerClient.DownloadFromContainer(containerID, docker.DownloadFromContainerOptions{
+		OutputStream: tempFile,
+		Path:         sourceDir,
+	})
+	if err != nil {
+		return err
+	}
+	if err := tempFile.Close(); err != nil {
+		return err
+	}
+
+	// Extract the created tar file to the destination directory
+	file, err := os.Open(tempFile.Name())
+	defer file.Close()
```

Comment:
this will fail if the err !=nil, because the "file" will be nil. move the defer below "if" ;-)